feature-libs/cart/base/components/add-to-cart/add-to-cart.component.ts

```diff
@@ -125,14 +129,25 @@ export class AddToCartComponent implements OnInit, OnDestroy {
       this.hasStock = true;
       this.cd.markForCheck();
     } else {
-      this.subscription = (
-        this.productListItemContext
-          ? this.productListItemContext.product$
-          : this.currentProductService.getProduct()
-      )
+      let productObservable$: Observable<Product | null>;
+
+      if (this.productListItemContext) {
+        productObservable$ = this.productListItemContext.product$;
+      } else {
+        if (this.featureToggles.showRealTimeStockInPDP) {
+          productObservable$ = this.currentProductService.getProduct(
```

Comment:
QUESTION: Do you think we need to add ProductScope.DETAILS along with ProductScope.UNIT so that we get unit extra along with other attributes as in the previous case?

Reply:
Checked in the network tab sapUnit is getting appended in the main product call so all the details along with sapUnit is loaded 

`{
   "availableForPickup" : false,
   "categories" : [ {
      "code" : "190100",
      "name" : "Snow Jackets",
      "url" : "/Categories/Snowwear-men/Snow-Jackets/c/190100"
   }, {
      "code" : "snow",
      "name" : "Snow",
      "url" : "/Collections/Snow/c/snow"
   }, {
      "code" : "Quiksilver",
      "image" : {
         "format" : "96Wx96H",
         "url" : "/medias/?context=bWFzdGVyfGltYWdlc3w1MzQ3fGltYWdlL2pwZWd8YURsbUwyaGtaUzg0TnprMk56Y3pNamcxT1RFNHwyZTM3NzY0ZWM2ZjE2NzdiZjJiYjFhMTFkYjdiOWYxYTcwYjAxMTc5YmYyODYxNzQ3MWZhZjgxOTdjOWQ2MTkw"
      },
      "name" : "Quiksilver",
      "url" : "/Brands/Quiksilver/c/Quiksilver"
   } ],
   "code" : "M_CR_1015",
   "configurable" : false,
   "description" : "",
   "multidimensional" : false,
   "name" : "Frozen redcurrant",
   "numberOfReviews" : 0,
   "price" : {
      "currencyIso" : "GBP",
      "formattedValue" : "GBP10.00",
      "priceType" : "BUY",
      "value" : 10.0
   },
   "priceRange" : {
   },
   "purchasable" : true,
   "sapUnit" : {
      "sapCode" : "ST"
   },
   "stock" : {
      "isValueRounded" : false,
      "stockLevel" : 249,
      "stockLevelStatus" : "inStock"
   },
   "summary" : "",
   "url" : "/Categories/Snowwear-men/Snow-Jackets/Frozen-redcurrant/p/M_CR_1015"
}`